compiler/ilgen/TypeDictionary.hpp

```diff
@@ -75,6 +75,53 @@ class IlType
    };
 
 
+/**
+ * @brief Convenience API for defining JitBuilder structs from C/C++ structs (PODs)
+ * 
+ * These macros simply allow the name of C/C++ structs and fields to be used to
+ * define JitBuilder structs. This can help ensure a consistent API between a
+ * VM struct and JitBuilder's representation of the same struct. Their definitions
+ * expand to calls to `TR::TypeDictionary` methods that create a representation
+ * corresponding to that specified type.
+ * 
+ * ## Usage
+ * 
+ * Given the following struct:
+ * 
+ * ```c++
+ * struct MyStruct {
+ *    int32_t id;
+ *    int32_t* val;
+ * };
+ * 
+ * a JitBuilder representation of this struct can be defined as follows:
+ * 
+ * ```c++
+ * TR::TypeDictionary d;
+ * d.DEFINE_STRUCT(MyStruct);
```

Comment:
Does the use of macros really make the code more readable? Its replacing one line with another one line of roughly similar character width. Also the macros are not undefined at the end of the header.